test/tests/functional/pbs_equiv_classes.py

```diff
@@ -0,0 +1,1052 @@
+
+        # Eat up all the resources
+        a = {'Resource_List.select': '1:ncpus=8'}
+        J = Job(TEST_USER, attrs=a)
+        self.server.submit(J)
+
+        jids1 = self.submit_jobs(3, a)
+
+        a = {'Resource_List.select': '1:ncpus=4'}
+        jids2 = self.submit_jobs(3, a)
+
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+
+        m = self.scheduler.log_match("Number of job equivalence classes: 2",
+                                     max_attempts=10, starttime=self.t)
+        self.assertTrue(m)
+
+    def test_select(self):
+        """
+        Test to see if jobs with select resources not in the resources line
+        fall into the same equivalence class
+        """
+        self.server.manager(MGR_CMD_CREATE, MGR_OBJ_RSC,
+                            {'type': 'long', 'flag': 'nh'}, id='foo')
+
+        # Eat up all the resources
+        a = {'Resource_List.select': '1:ncpus=8'}
+        J = Job(TEST_USER, attrs=a)
+        self.server.submit(J)
+
+        a = {'Resource_List.select': '1:ncpus=1:foo=4'}
+        jids1 = self.submit_jobs(3, a)
+
+        a = {'Resource_List.select': '1:ncpus=1:foo=8'}
+        jids2 = self.submit_jobs(3, a)
+
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+
+        # Two equivalence classes: one for the resource eating job and one
+        # for the other two jobs. While jobs have different amounts of
+        # the foo resource, foo is not on the resources line.
+        m = self.scheduler.log_match("Number of job equivalence classes: 2",
+                                     max_attempts=10, starttime=self.t)
+        self.assertTrue(m)
+
+    def test_place(self):
+        """
+        Test to see if jobs with different place statements
+        fall into the different equivalence classes
+        """
+        # Eat up all the resources
+        a = {'Resource_List.select': '1:ncpus=8'}
+        J = Job(TEST_USER, attrs=a)
+        self.server.submit(J)
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.place': 'free'}
+        jids1 = self.submit_jobs(3, a)
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.place': 'excl'}
+        jids2 = self.submit_jobs(3, a)
+
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+
+        # Three equivalence classes: one for the resource eating job and
+        # one for each place statement
+        m = self.scheduler.log_match("Number of job equivalence classes: 3",
+                                     max_attempts=10, starttime=self.t)
+        self.assertTrue(m)
+
+    def test_reslist1(self):
+        """
+        Test to see if jobs with resources in Resource_List that are not in
+        the sched_config resources line fall into the same equivalence class
```

Comment:
The tests are similar but not the same.  test_select tests resources within the select statement.  test_reslist1 tests resources outside of the select statement requested with qsub -l.  In the code there is a distinct difference between the two, so I wrote two tests.